src/go/supervisor/pkg/svc/supervisor.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright © 2021 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package svc
+
+import (
+	"fmt"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+
+	"github.com/rancher-sandbox/rancher-desktop/src/go/supervisor/pkg/port"
+)
+
+type supervisor struct {
```

Comment:
The code implementing the service really shouldn't be in the same package controlling the service (`ControlService` etc) — this is rather confusing, since they must run in different processes.

Reply:
The supervisor is not the service implementation but rather service definition (implements the service handler interface). The actual service implementation is being injected through `NewSupervisor(portServer *port.Server, logger debug.Log) *supervisor `